src/mococrw/pkcs11_engine.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2018 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#pragma once
+
+#include "openssl_wrap.h"
+#include "mococrw/asymmetric_crypto_ctx.h"
+
+namespace mococrw
+{
+
+/**
+ * A PKCS-11 Engine
+ *
+ */
+class Pkcs11Engine
```

Comment:
Maybe we want to be a bit more general here and have an engine class that is subclassed with a PKCS11Engine class?